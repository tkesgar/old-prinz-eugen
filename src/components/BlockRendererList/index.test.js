import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import BlockRendererList from './'

describe('BlockRendererList', () => {
  const style = 'ordered'
  const items = [
    'If you overwork yourself too much... you ~ will ~ die.',
    'Lucky ship? Are you mocking me?',
    'Operation Rheinübung? Looking back, that was how it all began for me.'
  ]

  const wrapper = mount(BlockRendererList, {
    localVue,
    propsData: {
      data: { style, items }
    }
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render matching number of items', () => {
    expect(wrapper.findAll('li').length).toBe(items.length)
  })

  test('should render <ol>', () => {
    expect(wrapper.find('ol').exists()).toBe(true)
  })
})

describe('BlockRendererList with style = unordered', () => {
  const style = 'unordered'
  const items = [
    'If you overwork yourself too much... you ~ will ~ die.',
    'Lucky ship? Are you mocking me?',
    'Operation Rheinübung? Looking back, that was how it all began for me.'
  ]

  const wrapper = mount(BlockRendererList, {
    localVue,
    propsData: {
      data: { style, items }
    }
  })

  test('should render <ul>', () => {
    expect(wrapper.find('ul').exists()).toBe(true)
  })
})

describe('BlockRendererList with empty items', () => {
  const style = 'unordered'
  const items = [
    'If you overwork yourself too much... you ~ will ~ die.',
    'Lucky ship? Are you mocking me?',
    'Operation Rheinübung? Looking back, that was how it all began for me.'
  ]

  const wrapper = mount(BlockRendererList, {
    localVue,
    propsData: {
      data: { style, items }
    }
  })

  test('should render <ul>', () => {
    expect(wrapper.find('ul').exists()).toBe(true)
  })
})
