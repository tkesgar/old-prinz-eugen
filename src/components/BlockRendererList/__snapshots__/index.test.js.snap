// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRendererList should match snapshot 1`] = `
<ol>
  <li>
    If you overwork yourself too much... you ~ will ~ die.
  </li>
  <li>
    Lucky ship? Are you mocking me?
  </li>
  <li>
    Operation Rhein√ºbung? Looking back, that was how it all began for me.
  </li>
</ol>
`;
